app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/agent/BusinessAccountingMethodController.scala

```diff
@@ -77,14 +77,14 @@ class BusinessAccountingMethodController @Inject()(businessAccountingMethod: Bus
           formWithErrors =>
             Future.successful(BadRequest(view(formWithErrors, id, isEditMode))),
           businessAccountingMethod =>
-            incomeTaxSubscriptionConnector.saveSubscriptionDetails(reference, businessAccountingMethodKey, businessAccountingMethod) map { _ =>
+            incomeTaxSubscriptionConnector.saveSubscriptionDetails(reference, businessAccountingMethodKey, businessAccountingMethod).map(_ =>
               (id, isEditMode, isEnabled(SaveAndRetrieve)) match {
-                case (Some(id), _, true) =>
-                  Redirect(routes.SelfEmployedCYAController.show(id, isEditMode = isEditMode))
+                case (Some(id), _, _) =>
```

Comment:
This case is missing the `true` value because we want to redirect the to the Self employed CYA page only when Save and Retrieve is enabled: `case (Some(id), _, _)` should be `case (Some(id), _, true)`.